cmake/CMakeLists.txt

```diff
@@ -441,30 +441,34 @@ if(HDF5_SUPPORT)
 
     if (HDF5_LIBRARY_DIR)
     	# result of the HDF5 ctest package
-    	
+
+
         if (WIN32)
-        	set(zlib_post lib)
+            # Windows
+            set(zlib_post zlib)
         	set(lib_ext lib)
+        elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            # macOS
+            set(zlib_post libz)
+            set(lib_ext dylib)
         else()
-        	set(lib_ext a)
+            # linux
+            set(zlib_post libz)
+            set(lib_ext so)
         endif()
 
-	if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-		set(debug_postfix "_debug")
-	endif() 
-        
-        SET(HDF5_LIBRARIES 
-            "${HDF5_LIBRARY_DIR}/libhdf5_cpp${debug_postfix}.${lib_ext}"
-            "${HDF5_LIBRARY_DIR}/libhdf5${debug_postfix}.${lib_ext}"
-            "${HDF5_LIBRARY_DIR}/libz${zlib_post}${debug_postfix}.${lib_ext}"
-            "${HDF5_LIBRARY_DIR}/libsz${debug_postfix}.${lib_ext}"
-            "${HDF5_LIBRARY_DIR}/libaec${debug_postfix}.${lib_ext}"
+        SET(HDF5_LIBRARIES
+            "${HDF5_LIBRARY_DIR}/libhdf5_cpp.${lib_ext}"
+            "${HDF5_LIBRARY_DIR}/libhdf5.${lib_ext}"
+            "${HDF5_LIBRARY_DIR}/${zlib_post}.${lib_ext}"
```

Comment:
I might have messed something up in the merge, but I do remember that I had to remove the permanent `libz` part in `libz${zlib_post}` and make it platform conditional so that it was able to find the right file on windows (which I believe is `zlib.lib` not `libz.lib`).

The `debug_postfix` is something relatively new in the v0.7.0 branch? So that could probably have been removed accidentally in my merge.

You are welcome to propose another solution. Or would you like me to propose something?

Reply:
Yes that's is the common, sensible name, but unfortunately the HDF5 install (that we use in build-deps and build-all to obtain zlib) deviates from that. So we have to account for that case. 

I guess we should replace all this with proper:

    FIND_LIBRARY(NAMES z libz libz_debug PATHS ... NO_DEFAULT_PATH)

That way we can also remove all the `lib_ext` stuff.

But an alternative is to extend the ugly trick we currently use to detect a conda build and put different logic there

https://github.com/IfcOpenShell/IfcOpenShell/blob/19fea3684f330747093c3072f6af7e9a9dc98883/cmake/CMakeLists.txt#L170-L173

~~~
 Directory of C:\Users\tkrij\Documents\IfcOpenShell\v0.7.0\_deps-vs2017-x64-installed\HDF5-1.8.22-win64\lib

07/10/2021  09:52 PM    <DIR>          .
07/10/2021  09:52 PM    <DIR>          ..
07/10/2021  09:40 PM           492,348 hdf5.lib
07/10/2021  09:41 PM           385,702 hdf5_cpp.lib
07/10/2021  09:44 PM            26,826 hdf5_hl.lib
07/10/2021  09:47 PM            11,752 hdf5_hl_cpp.lib
07/10/2021  09:40 PM            31,636 hdf5_tools.lib
07/10/2021  09:37 PM           118,598 libaec.lib
07/10/2021  09:43 PM        13,694,350 libhdf5.lib
07/10/2021  09:35 PM             2,985 libhdf5.settings
07/10/2021  09:47 PM         4,612,856 libhdf5_cpp.lib
07/10/2021  09:44 PM           565,292 libhdf5_hl.lib
07/10/2021  09:47 PM            38,210 libhdf5_hl_cpp.lib
07/10/2021  09:44 PM         1,086,940 libhdf5_tools.lib
07/10/2021  09:37 PM            20,496 libsz.lib
07/10/2021  09:38 PM           327,580 libzlib.lib
                                       ^^^^^^^^^^^
07/10/2021  09:52 PM    <DIR>          pkgconfig
              14 File(s)     21,415,571 bytes
               3 Dir(s)   1,548,943,360 bytes free
~~~